internal/provider/log_streams_data_source.go

```diff
@@ -0,0 +1,171 @@
+package provider
+
+import (
+	"context"
+	"terraform-provider-aembit/internal/provider/models"
+
+	"aembit.io/aembit"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &logStreamsDataSource{}
+	_ datasource.DataSourceWithConfigure = &logStreamsDataSource{}
+)
+
+// NewLogStreamsDataSource is a helper function to simplify the provider implementation.
+func NewLogStreamsDataSource() datasource.DataSource {
+	return &logStreamsDataSource{}
+}
+
+// logStreamsDataSource is the data source implementation.
+type logStreamsDataSource struct {
+	client *aembit.CloudClient
+}
+
+// Configure adds the logStream to the data source.
+func (d *logStreamsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	d.client = datasourceConfigure(req, resp)
+}
+
+// Metadata returns the data source type name.
+func (d *logStreamsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_log_streams"
+}
+
+// Schema defines the schema for the resource.
+func (d *logStreamsDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages a log stream.",
+		Attributes: map[string]schema.Attribute{
+			"log_streams": schema.ListNestedAttribute{
+				Description: "List of log streams.",
+				Computed:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						// ID field is required for Terraform Framework acceptance testing.
+						"id": schema.StringAttribute{
+							Description: "Unique identifier of the Log Stream.",
+							Computed:    true,
+						},
+						"name": schema.StringAttribute{
+							Description: "User-provided name of the Log Stream.",
+							Computed:    true,
+						},
+						"description": schema.StringAttribute{
+							Description: "User-provided description of the Log Stream.",
+							Computed:    true,
+						},
+						"is_active": schema.BoolAttribute{
+							Description: "Active/Inactive status of the Log Stream.",
+							Computed:    true,
+						},
+						"data_type": schema.StringAttribute{
+							Description: "Data type of Log Stream.",
+							Required:    true,
```

Comment:
```suggestion
							Computed:    true,
```
Note: I was doing this individually, but I'm going to stop on this one. All fields on data sources are read-only and therefore should be marked as `Computed`, not `Required`.